modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1317,152 +1379,153 @@ private RuntimeException convertThrowable(Throwable th) {
                     }
-                });
+                } catch (NodeStoppingException e) {
+                    // noop
+                } finally {
+                    if (raftGrpSvc != null) {
+                        raftGrpSvc.shutdown();
+                    }
+                }
 
-        setBaselineFut.join();
+                return true;
+            }
+
+            @Override
+            public void onError(@NotNull Throwable e) {
+                LOG.error("Error while processing pending assignments event", e);
+            }
+        });
     }
 
     /**
-     * Update raft groups of table partitions to new peers list.
+     * Extract table id from pending key of partition.
      *
-     * @param oldAssignments Old assignment.
-     * @param newAssignments New assignment.
-     * @param tblId Table ID.
-     * @return Future, which completes, when update finished.
+     * @param key Key.
+     * @return Table id.
      */
-    private CompletableFuture<Void> updateRaftTopology(
-            List<List<ClusterNode>> oldAssignments,
-            List<List<ClusterNode>> newAssignments,
-            UUID tblId) {
-        CompletableFuture<?>[] futures = new CompletableFuture<?>[oldAssignments.size()];
-
-        // TODO: IGNITE-15554 Add logic for assignment recalculation in case of partitions or replicas changes
-        // TODO: Until IGNITE-15554 is implemented it's safe to iterate over partitions and replicas cause there will
-        // TODO: be exact same amount of partitions and replicas for both old and new assignments
-        for (int i = 0; i < oldAssignments.size(); i++) {
-            final int p = i;
-
-            List<ClusterNode> oldPartitionAssignment = oldAssignments.get(p);
-            List<ClusterNode> newPartitionAssignment = newAssignments.get(p);
-
-            try {
-                futures[i] = raftMgr.changePeers(
-                        raftGroupName(tblId, p),
-                        oldPartitionAssignment,
-                        newPartitionAssignment
-                ).exceptionally(th -> {
-                    LOG.error("Failed to update raft peers for group " + raftGroupName(tblId, p)
-                            + "from " + oldPartitionAssignment + " to " + newPartitionAssignment, th);
-                    return null;
-                });
-            } catch (NodeStoppingException e) {
-                throw new AssertionError("Loza was stopped before Table manager", e);
-            }
-        }
+    private UUID extractTableId(ByteArray key) {
+        return UUID.fromString(key.toString().substring(PENDING_ASSIGNMENTS_PREFIX.length(), PENDING_ASSIGNMENTS_PREFIX.length() + 36));
+    }
 
-        return CompletableFuture.allOf(futures);
+    /**
+     * Extract partition number from the pending key of partition.
+     *
+     * @param key Key.
+     * @return Partition number.
+     */
+    private int extractPartitionNumber(ByteArray key) {
```

Comment:
What do you think about moving the both methods to a utility class?

Reply:
Done